plugin/src/test/kotlin/org/openrewrite/gradle/isolated/ResourceParserTest.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright ${year} the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.gradle.isolated
+
+import org.assertj.core.api.Assertions
+import org.gradle.api.Project
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito
+import org.openrewrite.gradle.DefaultRewriteExtension
+import org.openrewrite.java.internal.JavaTypeCache
+import kotlin.io.path.Path
+
+class ResourceParserTest {
+
+    @Test
+    fun `resource parser is excluding subprojects directories using the base dir`(
+    ) {
+        val project = Mockito.mock(Project::class.java)
```

Comment:
Gradle's `ProjectBuilder` is usually how I handle cases such as this.

Reply:
Thanks @shanman190 . I have added a new commit that uses `ProjectBuilder` instead of `Mockito`